Sources/Views/Cells/MessageCollectionViewCell.swift

```diff
@@ -144,4 +143,108 @@ open class MessageCollectionViewCell: UICollectionViewCell, CollectionViewReusab
     open func cellContentView(canHandle touchPoint: CGPoint) -> Bool {
         return false
     }
+
+    // MARK: - Origin Calculations
+
+    open func layoutAvatarView(with attributes: MessagesCollectionViewLayoutAttributes) {
+        guard attributes.avatarSize != .zero else { return }
+
+        var origin = CGPoint.zero
+
+        switch attributes.avatarPosition.horizontal {
+        case .cellLeading:
+            break
+        case .cellTrailing:
+            origin.x = attributes.frame.width - attributes.avatarSize.width
+        case .natural:
+            fatalError(MessageKitError.avatarPositionUnresolved)
+        }
+
+        switch attributes.avatarPosition.vertical {
+        case .cellTop:
+            break
+        case .cellBottom:
+            origin.y = attributes.frame.height - attributes.avatarSize.height
+        case .messageTop: // Needs messageContainerView frame to be set
+            origin.y = messageContainerView.frame.minY
+        case .messageBottom: // Needs messageContainerView frame to be set
+            origin.y = messageContainerView.frame.maxY - attributes.avatarSize.height
+        case .messageCenter: // Needs messageContainerView frame to be set
+            origin.y = messageContainerView.frame.midY - (attributes.avatarSize.height/2)
+        }
+
+        avatarView.frame = CGRect(origin: origin, size: attributes.avatarSize)
+    }
+
+    open func layoutMessageContainerView(with attributes: MessagesCollectionViewLayoutAttributes) {
+        guard attributes.messageContainerSize != .zero else { return }
+
+        var origin: CGPoint = .zero
```

Comment:
`var origin = CGPoint.zero` for consistency

Reply:
I think I'll flip flop them. I like explicitly typed better 👍